src/Microsoft.AspNetCore.Razor.Language/DefaultRazorIRLoweringPhase.cs

```diff
@@ -134,12 +134,19 @@ public override void VisitImportSpan(AddImportChunkGenerator chunkGenerator, Spa
 
             public override void VisitAddTagHelperSpan(AddTagHelperChunkGenerator chunkGenerator, Span span)
             {
-                _builder.Push(new DirectiveIRNode()
+                var directiveNode = new DirectiveIRNode()
                 {
                     Name = CSharpCodeParser.AddTagHelperDirectiveDescriptor.Directive,
                     Descriptor = CSharpCodeParser.AddTagHelperDirectiveDescriptor,
                     Source = BuildSourceSpanFromNode(span),
-                });
+                };
+
+                for (var i = 0; i < chunkGenerator.Diagnostics.Count; i++)
+                {
+                    directiveNode.Diagnostics.Add(chunkGenerator.Diagnostics[i]);
+                }
+
+                _builder.Push(directiveNode);
```

Comment:
The whole purpose of 'malformed' is that you don't process them by default. In the case you're suggesting, there's no value to be gained from 'malforming' since user code is doing it - (other than consistency).

Still if the user wanted to do it, it's trivial.